specification.md

```diff
@@ -6193,6 +6198,12 @@ In this use case, a third-party service may require information from the caller
 
 The specification doesn't define any specific variable within the `WORKFLOW` bucket, but it's considered a reserved keyword.
 
+### Naming Convention
+
+Identifiable components of a workflow definition, such as states, actions, branches, events and functions define a required non-null `name` property which follows the DNS label names as defined by [RFC 1123](https://datatracker.ietf.org/doc/html/rfc1123).
+
+In other words, `names` must be lowercased, start and end with an alphanumeric character, and contain only alphanumeric characters or '-' (dash) character.
```

Comment:
RFC 1123 does not include any lowercasing requirement; did you intend to make this a further restriction?

I think some text is also warranted regarding [RFC 5890 R-LDH labels](https://www.rfc-editor.org/rfc/rfc5890#section-2.3.1) (whose third and fourth characters are both "-") such as `xn--bcher-kva` (which is further an XN-label and valid A-label representing "bücher") and `we--like--it` (which is neither but is still subject to the "_MUST NOT be processed as ordinary LDH labels by IDNA-conforming programs_" restriction).

I don't think it matters whether they are allowed or rejected, but some software _does_ treat them specially so it's worth being explicit.

Reply:
@gibson042 I thought it did, or at least does it's k8s flavor: 

https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#:~:text=RFC%201123%20Label%20Names&text=in%20RFC%201123.-,This%20means%20the%20name%20must%3A,start%20with%20an%20alphanumeric%20character

Anyways, I do think adding the lowercase restriction, as i believe it could help ensure invariance, especially on exotic systems. 

It's not however a strong preference, and I'm obviously open to any and all opinion for and against ;)

As for the double dashes, I would venture to say it's safe to forbid its use altogether, like with the k8s regex I'm using in the PR. WDYT?

Reply:
:+1: